Deploying AKS environment configuration for production

shaheislamdfe opened the pull request:
### Context

<!-- Why are you making this change? -->

Adding configuration for s189 production environment AKS.

### Changes proposed in this pull request

Adding GHA workflow aswell as make target and tfvars configuration specific to production

### Guidance to review

Validate deployment on s189 aks cluster aswell as domains.

https://access-your-teaching-qualifications-production.teacherservices.cloud/qualifications/sign-in
https://check-a-teachers-record-production.teacherservices.cloud/

### Link to Trello card

https://trello.com/c/S5we2pWZ/2027-aytq-production-environment

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

saliceti commented on terraform/application/config/production.tfvars.json:
```diff
@@ -0,0 +1,11 @@
+    "cluster": "production",
+    "namespace": "tra-production",
+    "evidence_container_retention_in_days": 30,
+    "enable_monitoring": true,
```
Can we keep it false for the time being? We'll enable it closer to migration day

saliceti commented on terraform/application/config/production.tfvars.json:
```diff
@@ -0,0 +1,11 @@
+{
+    "cluster": "production",
+    "namespace": "tra-production",
+    "evidence_container_retention_in_days": 30,
```
TBC by developers

shaheislamdfe replied:
Have removed for time being

saliceti replied:
Developers have agreed with 30 days, this can be set

saliceti commented on terraform/application/config/production.tfvars.json:
```diff
@@ -0,0 +1,11 @@
+    "replicas": 2,
+    "azure_enable_high_availability": true,
+    "azure_enable_monitoring": true
+}
```
Can we use the same DB type as the existing? See https://github.com/DFE-Digital/access-your-teaching-qualifications/blob/main/terraform/workspace_variables/production.tfvars.json#L12

saliceti replied:
https://github.com/DFE-Digital/access-your-teaching-qualifications/blob/main/terraform/application/variables.tf#L33C11-L33C41 should be set to true so we can upload DB dumps

saliceti replied:
Could you please `"send_traffic_to_maintenance_page": false`. It's useful to have it here in case of a DR scenario

saliceti replied:
Can we set account_replication_type to GRS. It's safer for prod

shaheislam committed:
Add production environment for AKS

Add production target for makefile, aswell as config values, tfvar values specific to production. Adding GHA workflows also.
